From d8f10be2191bd3fdf257fd6a6467163f62c0f927 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:04:25 -0400
Subject: [powerpc] Align p_toc

Message-id: <1410545655-205645-37-git-send-email-dzickus@redhat.com>
Patchwork-id: 94074
O-Subject: [RHEL7 PATCH 036/626] powerpc: Align p_toc
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 5b63fee1fe5d10c397d740ffef4d576a196ed72d
Author: Anton Blanchard <anton@samba.org>
Date:   Wed Aug 7 02:01:18 2013 +1000

    powerpc: Align p_toc

    p_toc is an 8 byte relative offset to the TOC that we place in the
    text section. This means it is only 4 byte aligned where it should
    be 8 byte aligned. Add an explicit alignment.

    Signed-off-by: Anton Blanchard <anton@samba.org>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kernel/head_64.S b/arch/powerpc/kernel/head_64.S
index cdd5f2e..2ae41ab 100644
--- a/arch/powerpc/kernel/head_64.S
+++ b/arch/powerpc/kernel/head_64.S
@@ -706,6 +706,7 @@ _GLOBAL(relative_toc)
  mtlr r0
  blr
 
+.balign 8
 p_toc: .llong __toc_start + 0x8000 - 0b
 
 /*
-- 
1.7.1